src/libs/Middleware/Pagination.ts

```diff
@@ -0,0 +1,66 @@
+// TODO: Is this a legit use case for exposing `OnyxCache`, or should we use `Onyx.connect`?
+import OnyxCache from 'react-native-onyx/dist/OnyxCache';
+import {READ_COMMANDS, WRITE_COMMANDS} from '@libs/API/types';
+import * as ReportActionsUtils from '@libs/ReportActionsUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ReportActions, ReportMetadata, Response} from '@src/types/onyx';
+import type Middleware from './types';
+
+function getReportActions(response: Response, reportID: string) {
+    return response?.onyxData?.find((data) => data.key === `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`)?.value as ReportActions | undefined;
+}
+
+const Pagination: Middleware = (requestResponse, request) => {
+    if (request.command === WRITE_COMMANDS.OPEN_REPORT || request.command === READ_COMMANDS.GET_OLDER_ACTIONS || request.command === READ_COMMANDS.GET_NEWER_ACTIONS) {
+        return requestResponse.then((response) => {
+            if (!response?.onyxData) {
+                return Promise.resolve(response);
+            }
+
+            const reportID = request.data?.reportID as string | undefined;
+            if (!reportID) {
+                // TODO: Should not happen, should we throw?
+                return Promise.resolve(response);
+            }
+
+            const reportActionID = request.data?.reportActionID as string | undefined;
+
+            // Create a new page based on the response actions.
+            const pageReportActions = getReportActions(response, reportID);
+            const pageSortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(pageReportActions, true);
```

Comment:
I am using `getSortedReportActionsForDisplay` since this is what will be used in `sortedReportActions` in `ReportScreen`. I am pretty sure this is correct.